lightning/src/ln/channelmanager.rs

```diff
@@ -6294,20 +6315,8 @@ where
 				err: format!("Payment with intercept id {} not found", log_bytes!(intercept_id.0))
 			})?;
 
-		if let PendingHTLCRouting::Forward { short_channel_id, incoming_cltv_expiry, .. } = payment.forward_info.routing {
-			let htlc_source = HTLCSource::PreviousHopData(HTLCPreviousHopData {
-				short_channel_id: payment.prev_short_channel_id,
-				user_channel_id: Some(payment.prev_user_channel_id),
-				outpoint: payment.prev_funding_outpoint,
-				channel_id: payment.prev_channel_id,
-				counterparty_node_id: Some(payment.prev_counterparty_node_id),
-				htlc_id: payment.prev_htlc_id,
-				incoming_packet_shared_secret: payment.forward_info.incoming_shared_secret,
-				phantom_shared_secret: None,
-				blinded_failure: payment.forward_info.routing.blinded_failure(),
-				cltv_expiry: incoming_cltv_expiry,
-			});
-
+		if let PendingHTLCRouting::Forward { short_channel_id, .. } = payment.forward_info.routing {
+			let htlc_source = HTLCSource::PreviousHopData(payment.htlc_previous_hop_data());
```

Comment:
Phantom secret was set to None always, where as now it's taken from `forward_info`. No problem? Also at the last change in this file.

Reply:
It should always be `None` here I believe regardless, since intercept payments and phantom payments both lean on special scids I don't think it would be possible to have a payment be both intercept and phantom.